core/property/registry.go

```diff
@@ -2,31 +2,66 @@ package property
 
 import (
 	"fmt"
-	"sync"
-
+	"github.com/TIBCOSoftware/flogo-lib/config"
 	"github.com/TIBCOSoftware/flogo-lib/core/data"
 	"github.com/TIBCOSoftware/flogo-lib/logger"
+	"reflect"
+	"regexp"
+	"sync"
 )
 
 var (
-	props = make(map[string]string)
-	mut   = sync.RWMutex{}
+	props    = make(map[string]interface{})
```

Comment:
Same here, I would keep it string. It simplifies everything and pushes complexity to the client, that has all the information about the casting.